app/controllers/jobs_controller.rb

```diff
@@ -159,17 +183,24 @@ def formatRawJob(rjob, is_live)
     job[:rjob] = rjob
     job[:id] = rjob["id"]
     job[:name] = rjob["name"]
+    
+    if rjob["notifyURL"] then
+      params =  rjob["notifyURL"].split('/')
+      job[:submission] = params[-2]
+      job[:assessment] = params[-4]
+      job[:course] = params[-6]
+    end
 
-    # Determine whether to expose the job name (which contains an AndrewID).
+    # Determine whether to expose the job name.
     if !@cud.user.administrator?  then
-      if !@cud.instructor? then 
+      if !@cud.instructor? then
         # Students can see only their own job names
         if !job[:name][@cud.user.email] then
           job[:name] = "*"
         end
       else
         # Instructors can see only their course's job names
-        if !rjob["notifyURL"] then
+        if !rjob["notifyURL"] or !(job[:course].eql? @cud.course.id.to_s) then
```

Comment:
job[:course] is a string here since we're just getting it by regexping the notifyUrl.